go.mod

```diff
@@ -3,42 +3,42 @@ module sigs.k8s.io/gcp-compute-persistent-disk-csi-driver
 go 1.19
 
 require (
-	cloud.google.com/go/compute v1.7.0
-	cloud.google.com/go/kms v1.4.0
+	cloud.google.com/go/compute/metadata v0.2.3
+	cloud.google.com/go/kms v1.6.0
 	github.com/GoogleCloudPlatform/k8s-cloud-provider v1.18.0
 	github.com/container-storage-interface/spec v1.6.0
 	github.com/google/uuid v1.3.0
 	github.com/kubernetes-csi/csi-proxy/client v1.1.1
 	github.com/kubernetes-csi/csi-test/v4 v4.4.0
 	github.com/onsi/ginkgo/v2 v2.7.1
 	github.com/onsi/gomega v1.25.0
-	golang.org/x/oauth2 v0.0.0-20220722155238-128564f6959c
-	golang.org/x/sys v0.4.0
-	google.golang.org/api v0.86.0
-	google.golang.org/genproto v0.0.0-20220720214146-176da50484ac
-	google.golang.org/grpc v1.48.0
-	google.golang.org/protobuf v1.28.0
+	golang.org/x/oauth2 v0.5.0
+	golang.org/x/sys v0.5.0
+	google.golang.org/api v0.111.0
+	google.golang.org/genproto v0.0.0-20230223222841-637eb2293923
+	google.golang.org/grpc v1.53.0
+	google.golang.org/protobuf v1.28.1
 	gopkg.in/gcfg.v1 v1.2.3
+	k8s.io/api v0.24.1
 	k8s.io/apimachinery v0.24.1
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
```

Comment:
You may be able to get rid of the replace statement if you put 0.24.1 here

Reply:
I already tried that and wrestled with this for a while:

```
$> go mod tidy
go: finding module for package k8s.io/api/settings/v1alpha1
go: finding module for package k8s.io/api/batch/v2alpha1
go: finding module for package github.com/googleapis/gnostic/OpenAPIv2
go: finding module for package k8s.io/api/auditregistration/v1alpha1
sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration imports
        k8s.io/client-go/kubernetes imports
        k8s.io/client-go/discovery imports
        github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest found (v0.6.9), but does not contain package github.com/googleapis/gnostic/OpenAPIv2
sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration imports
        k8s.io/client-go/kubernetes imports
        k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1 imports
        k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.26.2), but does not contain package k8s.io/api/auditregistration/v1alpha1
sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration imports
        k8s.io/client-go/kubernetes imports
        k8s.io/client-go/kubernetes/typed/batch/v2alpha1 imports
        k8s.io/api/batch/v2alpha1: module k8s.io/api@latest found (v0.26.2), but does not contain package k8s.io/api/batch/v2alpha1
sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/k8s-integration imports
        k8s.io/client-go/kubernetes imports
        k8s.io/client-go/kubernetes/typed/settings/v1alpha1 imports
        k8s.io/api/settings/v1alpha1: module k8s.io/api@latest found (v0.26.2), but does not contain package k8s.io/api/settings/v1alpha1
```

Reply:
I believe that means that some of our dependencies need us to specify the version via replace.

Reply:
Looks like `v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible` is arbitrary and doesn't show up in our go.sum file. Pulling the latest version is just a way for `go mod tidy` to satisfy the require directive syntax.